### DISCUSSION TITLE
Redrawing specific paths on a new page

### DISCUSSION THREAD

**Discussion by qwertynik** [pymupdf/PyMuPDF#1730] [Looking for help]
Hello,

Had been using the draw_* methods listed [here](https://pymupdf.readthedocs.io/en/latest/page.html#modifying-pages) to draw both new and existing paths in a PDF. However, was looking for ways to draw a path on a new page as is, without having to choose which draw_* method to use. This is so that no path is missed or incorrectly drawn in the new page.

So for instance, a method like draw_path(path) which will draw the path as is.

Can you recommend some approaches to accomplish the same @JorjMcKie?

List of available draw methods for easier reference:

```
draw_bezier()
draw_circle()
draw_curve()
draw_line()
draw_oval()
draw_polyline()
draw_quad()
draw_rect()
draw_sector()
draw_squiggle()
draw_zigzag()
```

Thanks!

**Comment by qwertynik** [comment]
@JorjMcKie Can you share your insights here?

**Comment by JorjMcKie** [comment]
There is no support for this in PyMuPDF - and will never be as far as I can tell.

**Reply by qwertynik** [reply]
@JorjMcKie 
Hmm, so until then one should pick and choose the appropriate draw_* methods. Have you come across any case where the existing draw_* methods did not handle a use-case?

**Reply by qwertynik** [reply]
>There is no support for this in PyMuPDF - and will never be as far as I can tell.

@JorjMcKie Wondering why this was mentioned. Because, when going through `mutool trace` command the output for vectors appeared 'straightforward' with instructions like, moveTo, lineTo, etc. So, is it because of some other technical complexity or just that this is not a priority?

**Reply by JorjMcKie** [reply]
> because of some other technical complexity or just that this is not a priority?

Both.
Don't let yourself be lulled by looking at trivial examples: Sophisticated cases can have arbitrary stacking levels of drawings (and clippings - which are ignored completely by the method) above each other, with alternating uses of rules (even-odd, nonzero-winding) and what not else.

**Reply by qwertynik** [reply]
Yes @JorjMcKie, from your earlier mention had been aware of the complexity that can be introduced by even-odd and nonzero-winding rules - haven't seen a working example though.

>Sophisticated cases can have arbitrary stacking levels of drawings (and clippings - which are ignored completely by the method) above each other

Can you elaborate on stacking levels of drawing? From what I understand this is pointing to an overlay of drawings - which does not 'appear' complex.

**Reply by JorjMcKie** [reply]
examples are paths containing sub-paths containing sub-paths containing ... mixed with clipping paths (which can make invisible parts of whatever sub-path(s) follow) ...

**Reply by JorjMcKie** [reply]
all this is not supported by neither `page.get_drawigs()` nor by the `.draw_*` methods.

**Reply by JorjMcKie** [reply]
also please look at issue #1518

**Comment by JorjMcKie** [comment]
Picking the right draw method is a no-brainer - by all means! There is a sample script in the utilities repo which does that using less than a handful code lines.
I mentioned before that there are principal limits as to how true such a reproduction can be - especially as per fill color appearance: think again about my mentioning the even-odd / nonzero-winding rules.

**Reply by qwertynik** [reply]
> Picking the right draw method is a no-brainer - by all means! There is a sample script in the utilities repo which does that using less than a handful code lines.

Yes, with some insights from one of the scripts, had built a script - however, it does not support all the shapes.
Since, as you say, it is a no-brainer, will look for ways to support other shape types.
